Client/Frontend/Sync/BraveCore/BraveSyncAPIExtensions.swift

```diff
@@ -8,14 +8,44 @@ import BraveCore
 import BraveShared
 import Shared
 
+public struct BraveSyncDevice: Codable {
+  let chromeVersion: String
+  let hasSharingInfo: Bool
+  let id: String
+  let guid: String
+  let isCurrentDevice: Bool
+  let supportsSelfDelete: Bool
+  let lastUpdatedTimestamp: TimeInterval
+  let name: String?
+  let os: String
+  let sendTabToSelfReceivingEnabled: Bool
+  let type: String
+}
+
+private let log = Logger.browserLogger
+
 extension BraveSyncAPI {
 
   public static let seedByteLength = 32
-
+  
   var isInSyncGroup: Bool {
     return Preferences.Chromium.syncEnabled.value
   }
 
+  var isSendTabToSelfVisible: Bool {
+    guard let json = getDeviceListJSON(), let data = json.data(using: .utf8) else {
```

Comment:
Do you need this? Usually we pull devices only when displaying the entire list of devices on the chain (setting screen).

Reply:
Yes this is totally needed in order to hide and show "Send To Your Devices: action properly.This is helping to check if we have other devices in the chain so we will not show the send action prematurely.